app/upgrades/v16/concentrated_pool.go

```diff
@@ -0,0 +1,154 @@
+package v16
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authkeeper "github.com/cosmos/cosmos-sdk/x/auth/keeper"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+
+	"github.com/osmosis-labs/osmosis/osmoutils"
+	"github.com/osmosis-labs/osmosis/v15/app/keepers"
+	clmodel "github.com/osmosis-labs/osmosis/v15/x/concentrated-liquidity/model"
+	gammkeeper "github.com/osmosis-labs/osmosis/v15/x/gamm/keeper"
+	gammtypes "github.com/osmosis-labs/osmosis/v15/x/gamm/types"
+	incentiveskeeper "github.com/osmosis-labs/osmosis/v15/x/incentives/keeper"
+	incentivestypes "github.com/osmosis-labs/osmosis/v15/x/incentives/types"
+	poolincentiveskeeper "github.com/osmosis-labs/osmosis/v15/x/pool-incentives/keeper"
+	"github.com/osmosis-labs/osmosis/v15/x/poolmanager"
+	poolmanagertypes "github.com/osmosis-labs/osmosis/v15/x/poolmanager/types"
+)
+
+// getGaugesForCFMMPool returns the gauges associated with the given CFMM pool ID, by first retrieving
+// the lockable durations for the pool, then using them to query the pool incentives keeper for the
+// gauge IDs associated with each duration, and finally using the incentives keeper to retrieve the
+// actual gauges from the retrieved gauge IDs.
+func getGaugesForCFMMPool(ctx sdk.Context, incentivesKeeper incentiveskeeper.Keeper, poolIncentivesKeeper poolincentiveskeeper.Keeper, poolId uint64) ([]incentivestypes.Gauge, error) {
+	lockableDurations := poolIncentivesKeeper.GetLockableDurations(ctx)
+	cfmmGaugeIds := make([]uint64, 0, len(lockableDurations))
+	for _, duration := range lockableDurations {
+		gaugeId, err := poolIncentivesKeeper.GetPoolGaugeId(ctx, poolId, duration)
+		if err != nil {
+			return nil, err
+		}
+		cfmmGaugeIds = append(cfmmGaugeIds, gaugeId)
+	}
+
+	cfmmGauges, err := incentivesKeeper.GetGaugeFromIDs(ctx, cfmmGaugeIds)
+	if err != nil {
+		return nil, err
+	}
+
+	return cfmmGauges, nil
+}
+
+// createConcentratedPoolFromCFMM creates a new concentrated liquidity pool with the desiredDenom0 token as the
+// token 0, links it with an existing CFMM pool, and returns the created pool.
+// It first creates a module account for pool manager module, and then creates the pool from
+// that module account.
+// Returns error if desired denom 0 is not in associated with the CFMM pool.
+// Returns error if CFMM pool does not have exactly 2 denoms.
+// Returns error if pool creation fails.
+func createConcentratedPoolFromCFMM(ctx sdk.Context, cfmmPoolIdToLinkWith uint64, desiredDenom0 string, accountKeeper authkeeper.AccountKeeper, gammKeeper gammkeeper.Keeper, poolmanagerKeeper poolmanager.Keeper) (poolmanagertypes.PoolI, error) {
+	cfmmPool, err := gammKeeper.GetCFMMPool(ctx, cfmmPoolIdToLink)
+	if err != nil {
+		return nil, err
+	}
+
+	poolCreatorAddress := authtypes.NewModuleAddress(poolmanagertypes.ModuleName)
+	if err := osmoutils.CreateModuleAccount(ctx, accountKeeper, poolCreatorAddress); err != nil {
+		return nil, err
+	}
```

Comment:
Do we need to create a new module account or does poolManager already have an existing module account?